Services/AssessmentQuestion/interfaces/interface.ilAsqQuestionSolution.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/* Copyright (c) 1998-2013 ILIAS open source, Extended GPL, see docs/LICENSE */
+
+/**
+ * Interface ilAsqQuestionSolution
+ *
+ * @author    Björn Heyser <info@bjoernheyser.de>
+ * @version    $Id$
+ *
+ * @package    Services/AssessmentQuestion
+ */
+interface ilAsqQuestionSolution
+{
+	/**
+	 * @param integer $solutionId
+	 * @return void
+	 */
+	public function setSolutionId($solutionId);
+	
+	/**
+	 * @return integer
+	 */
+	public function getSolutionId();
+	
+	/**
+	 * @param integer $questionId
+	 * @return void
+	 */
+	public function setQuestionId($questionId);
+	
+	/**
+	 * @return integer
+	 */
+	public function getQuestionId();
+	
+	/**
+	 * @return void
+	 */
+	public function load();
+	
+	/**
+	 * @return void
+	 */
+	public function save();
+	
+	/**
+	 * @param \Psr\Http\Message\ServerRequestInterface $request
+	 * @return void
+	 */
+	public function initFromServerRequest(\Psr\Http\Message\ServerRequestInterface $request);
```

Comment:
Is it really the job of ilAsqQuestionSolution to understand requests? Maybe remove this dependency here.

Reply:
ilAsqQuestionSolution is the object that will understand the post structure. wen can switch to passing $_POST, but i kept in mind that we want to get rid of direct access to any super global. I dont know, but @mjansen and me did think this is a quite good idea.